pkg/scheduler/core/generic_scheduler.go

```diff
@@ -393,7 +412,19 @@ func (g *genericScheduler) findNodesThatFit(pod *v1.Pod, nodes []*v1.Node) ([]*v
 				predicateResultLock.Unlock()
 			}
 		}
-		workqueue.Parallelize(16, len(nodes), checkNode)
+		allNodes := int32(g.cache.NodeTree().NumNodes)
+		numNodesToFind := g.numFeasibleNodesToFind(allNodes)
+		numNodesProcessed := int32(0)
+		for numNodesProcessed < allNodes {
```

Comment:
Why not do it all at once until filteredLen >= numNodesToFind?

Reply:
once you send work to `parallelize` you cannot stop in the middle.

Reply:
Adding a parameter to parallelize to control is a solution, or just redefining an interface to implement this functionality.

Reply:
Parallelize is a part of a client-go library. We cannot change its parameters, but I agree that having another Parallelize function, for example, ParallelizeUntil(..., condition) would be useful. That should be done as a separate PR though. Do you think you can add one that can be used here?

Reply:
Ok, when this PR is merged I will help implement it. :)